File: src/main/java/ch/qligier/jetbrains/plugin/fss/igini/language/psi/impl/IniPsiImplUtil.java
// Copyright 2024 Quentin Ligier. Use of this source code is governed by the MIT license.

package ch.qligier.jetbrains.plugin.fss.igini.language.psi.impl;

import ch.qligier.jetbrains.plugin.fss.igini.language.psi.*;
import com.intellij.lang.ASTNode;
import com.intellij.psi.util.PsiTreeUtil;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

/**
 * jetbrains-plugin-fss
 *
 * @author Quentin Ligier
 **/
public class IniPsiImplUtil {

    public static @NotNull List<IniSection> getSections(final IniFile file) {
        return PsiTreeUtil.getChildrenOfTypeAsList(file, IniSection.class);
    }

    public static @NotNull List<IniProperty> getProperties(final IniFile file) {
        return PsiTreeUtil.getChildrenOfTypeAsList(file, IniProperty.class);
    }

    public static @Nullable IniKey getKeyElement(final IniProperty element) {
        return PsiTreeUtil.getChildOfType(element, IniKey.class);
    }

    public static @Nullable String getKeyName(final IniProperty element) {
        final ASTNode keyNode = element.getNode().findChildByType(IniTypes.KEY);
        if (keyNode != null) {
            return keyNode.getText();
        } else {
            return null;
        }
    }

    public static @Nullable String getSectionName(final IniSection element) {
        final ASTNode keyNode = element.getNode().findChildByType(IniTypes.SECTION_NAME);
        if (keyNode != null) {
            return keyNode.getText();
        } else {
            return null;
        }
    }

    public static @NotNull List<IniProperty> getProperties(final IniSection section) {
        return PsiTreeUtil.getChildrenOfTypeAsList(section, IniProperty.class);
    }
}

File: src/main/kotlin/ch/qligier/jetbrains/plugin/fss/ig_ini/language/psi/impl/IniPsiImplUtil.kt
/*
 * Copyright 2025 Quentin Ligier. Use of this source code is governed by the MIT license.
 */
package ch.qligier.jetbrains.plugin.fss.ig_ini.language.psi.impl

import ch.qligier.jetbrains.plugin.fss.ig_ini.language.psi.IniFile
import ch.qligier.jetbrains.plugin.fss.ig_ini.language.psi.IniKey
import ch.qligier.jetbrains.plugin.fss.ig_ini.language.psi.IniProperty
import ch.qligier.jetbrains.plugin.fss.ig_ini.language.psi.IniSection
import ch.qligier.jetbrains.plugin.fss.ig_ini.language.psi.IniTypes
import com.intellij.psi.util.PsiTreeUtil

object IniPsiImplUtil {
    @JvmStatic
    fun getSections(file: IniFile?): MutableList<IniSection> {
        return PsiTreeUtil.getChildrenOfTypeAsList(file, IniSection::class.java)
    }

    @JvmStatic
    fun getProperties(file: IniFile?): MutableList<IniProperty> {
        return PsiTreeUtil.getChildrenOfTypeAsList(file, IniProperty::class.java)
    }

    @JvmStatic
    fun getKeyElement(element: IniProperty?): IniKey? {
        return PsiTreeUtil.getChildOfType(element, IniKey::class.java)
    }

    @JvmStatic
    fun getKeyName(element: IniProperty): String? = element.node.findChildByType(IniTypes.KEY)?.text

    @JvmStatic
    fun getSectionName(element: IniSection): String? = element.node.findChildByType(IniTypes.SECTION_NAME)?.text

    @JvmStatic
    fun getProperties(section: IniSection?): MutableList<IniProperty> {
        return PsiTreeUtil.getChildrenOfTypeAsList(section, IniProperty::class.java)
    }
}
